### ISSUE TITLE
Train gets stuck after consecutive turns in oposite direction

### ISSUE THREAD

**Issue by peter-vasut** [orwell96/advtrains#57]
Hi, I'm having following problem. It is probably bug.

**Steps to reproduce:**

- place few straight rails
- place one 30° right turn
- place one 30° left turn
- place few straight rails
- run steam locomotive from one end

**Behavior:**

- Train stops on turn.
- I get message saying _Train 4876 is off track. Trying to drive further out. Velocity clipped to 0_

**Expected behavior:**

- Train should continue and stop at the end of track.

**Comment by orwell96** [comment]
Trains shouldn't run off tracks. So they stop (or start to brake heavily) as soon as they reach an end. Due to internal mechanics, they do so 1 rail before the actual 'end'.
In the case you describe, the behavior is the same like when driving at the end of a straight rail: the train stops 1 rail before the end. Because there's 2/1 space between the rails (knight's move), the train stops somewhat far from the actual rail end. That's not a bug.
However, I could extend the limit how far trains can approach rail ends.
BTW: If you drive a fast train towards a rail end, it may roll over that end. If the last rail is a curve, it's ignored. That's not a bug too, but just internal mechanics.

**Comment by peter-vasut** [comment]
Its my fault. Example I mentioned works OK. I just needed to create longer tracks. It was probably coincidence, that it stopped on those consecutive turns, and when I was testing it on small test track, I made short enough track that it looked like it is again stopped because of turns. Probably I have problem with [crossing](http://imgur.com/a/L0VHC) next to that part. I'm sorry for bothering you.
